src/module.ts

```diff
@@ -121,11 +125,17 @@ export default defineNuxtModule<ModuleOptions>({
           }
         })
       }
+      let swGenerated = false
       nuxt.hook('nitro:init', (nitro) => {
         nitro.hooks.hook('rollup:before', async () => {
+          swGenerated = true
           await resolveVitePluginPWAAPI()?.generateSW()
         })
       })
+      nuxt.hook('close', async () => {
+        if (!swGenerated)
+          await resolveVitePluginPWAAPI()?.generateSW()
+      })
```

Comment:
We could also do it like this:

```suggestion
      if (nuxt.options._generate) {
        nuxt.hook('close', async () => {
          await resolveVitePluginPWAAPI()?.generateSW()
        })
      }
```